src/js/components/ValidationProgress.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { PropTypes } from 'react'
+
+const ValidationProgress = (props) => {
+
+  const code = props.status.code
+
+  let uploadComplete = null
+  let parsingStarted = null
+  let parsingComplete = null
+  let validationStarted = null
+  let validationComplete = null
+
```

Comment:
Validation progress is only rendered when the upload completes and there is a separate progress tracker for the upload.

Also, I'd be fine changing the design of this component but maybe in another PR (this PR purposely has minimal style)